vizro-core/docs/pages/user_guides/actions.md

```diff
@@ -384,9 +384,172 @@ The order of action execution is guaranteed, and the next action in the list wil
+                        vm.Action(function=filter_interaction(targets=["scatter_chart_2"])),
+                        vm.Action(
+                            function=my_custom_action(),
+                            inputs=["scatter_chart.clickData"],
+                            outputs=["card_id.children"]
+                        ),
+                    ],
+                ),
+                vm.Card(id="card_id", text="### No data clicked."),
+                vm.Graph(
+                    id="scatter_chart_2",
+                    figure=px.scatter(df, x="sepal_length", y="petal_width", color="species"),
+                ),
+            ],
+            controls=[vm.Filter(column="species", selector=vm.Dropdown(title="Species"))],
+        )
+
+        dashboard = vm.Dashboard(pages=[page])
+
+        Vizro().build(dashboard).run()
+        ```
+    === "app.yaml"
+        ```yaml
+        # Custom action are currently only possible via python configuration
+        ```
+
+#### Custom action return value
+The return value of the custom action function is propagated to the dashboard components that are defined in the `outputs` argument of the [`Action`][vizro.models.Action] model.
+If there is a single `output` defined, the function return value is directly assigned to the component property.
+If there are multiple `outputs` defined, the return value is iterated and assigned to the respective component properties, in line with Python's flexibility in managing multiple return values.
+
+!!! tip
+    There is an option to match return values to multiple `outputs` in a desired order (which could be quite useful if the action has many outputs) by using [`collections.abc.namedtuple`](https://docs.python.org/3/library/collections.html#namedtuple-factory-function-for-tuples-with-named-fields).
+    The easies way to achieve it is to create a dictionary object where keys represent outputs in the following format: `"<component_id>_<property>"` and dictionary values represent the return value to be applied on the key output. Then, the dictionary can be easily converted to a namedtuple object and returned from the custom action function.
+    Here is an example of what the custom action function from the previous example would look like if we wanted to match the return value with multiple `outputs` in the desired order:
+    ```py
+    from collections import namedtuple
+    @capture("action")
+    def my_custom_action(scatter_chart_clickData: dict = None):
+        """Custom action."""
+        if scatter_chart_clickData:
+            return_value = {
+                "card_id_children": f'Scatter chart clicked data:\n### Species: "{scatter_chart_clickData["points"][0]["customdata"][0]}"'}
+            }
+            return namedtuple("x", return_value.keys())(*return_value.values())
+        return_value = {"card_id_children": "### No data clicked."}
+        return namedtuple("x", return_value.keys())(*return_value.values())
+    ```
```

Comment:
If we need to add another output in the example here, maybe I can add a "card_id_style" that will apply a specific card background colour depending on which "specis" is clicked in the original chart.

Reply:
Oh sorry, I didn't see this idea when I wrote another example here that just has two output cards. Feel free to change if you'd like a nicer example - I don't mind 🙂